ospfd/ospf_vty.c

```diff
@@ -4717,34 +4717,45 @@ static int show_ip_ospf_neighbor_int_common(struct vty *vty, struct ospf *ospf,
 
 DEFUN (show_ip_ospf_neighbor_int,
        show_ip_ospf_neighbor_int_cmd,
-       "show ip ospf neighbor IFNAME [json]",
+       "show ip ospf [vrf <NAME>] neighbor IFNAME [json]",
        SHOW_STR
        IP_STR
        "OSPF information\n"
+       VRF_CMD_HELP_STR
        "Neighbor list\n"
        "Interface name\n"
        JSON_STR)
 {
 	struct ospf *ospf;
 	int idx_ifname = 4;
+	int idx_vrf;
 	bool uj = use_json(argc, argv);
-	struct listnode *node = NULL;
 	int ret = CMD_SUCCESS;
 	struct interface *ifp = NULL;
+	char *vrf_name = NULL;
+	vrf_id_t vrf_id = VRF_DEFAULT;
+	struct vrf *vrf = NULL;
+
+	if (argv_find(argv, argc, "vrf", &idx_vrf))
+		vrf_name = argv[idx_vrf + 1]->arg;
+	if (vrf_name == NULL || strmatch(vrf_name, VRF_DEFAULT_NAME))
+		vrf_name = NULL;
+	if (vrf_name)
+		vrf = vrf_lookup_by_name(vrf_name);
+	if (vrf)
+		vrf_id = vrf->vrf_id;
+	ospf = ospf_lookup_by_vrf_id(vrf_id);
 
 	if (!uj)
 		show_ip_ospf_neighbour_header(vty);
 
-	ifp = if_lookup_by_name_all_vrf(argv[idx_ifname]->arg);
-	for (ALL_LIST_ELEMENTS_RO(om->ospf, node, ospf)) {
-		if (!ospf->oi_running)
-			continue;
-		if (!ifp || ifp->vrf_id != ospf->vrf_id)
-			continue;
-		ret = show_ip_ospf_neighbor_int_common(vty, ospf, idx_ifname,
-						       argv, uj, 0);
-	}
-
+	if (!argv_find(argv, argc, "IFNAME", &idx_ifname))
+		return ret;
```

Comment:
`IFNAME` is a mandatory parameter, I don't think we need to check if `argv_find()` succeeded or not here.

Also, it would be good to initialize `idx_ifname` to zero now that we're using `argv_find()`.